utils/cache/cache.h

```diff
@@ -75,7 +75,7 @@ status_t cache_del(const char* const key);
  * - SC_OK on success
  * - non-zero on error
  */
-status_t cache_get(const char* const key, char* res);
+status_t cache_get(const char* const key, char** res);
```

Comment:
Why did you change the prototype of `cache_get`? I suppose this pull request should not change the public interface.

Reply:
The string that MAM is going to buffer is variable length.  The responded object, res, in `cache_get` was allocated first. It would cause some problems under MAM scenario

Reply:
I see. However, it is really a bad manner to violate the public interface.

Reply:
Or we should set up maximum size for a MAM request

Reply:
> Or we should set up maximum size for a MAM request

It sounds reasonable at the moment.

Reply:
I will open another issue for discussing the limitation of MAM request